File: pbark/pbark/parser/keywords/print_keyword_groups.py
from __future__ import annotations

from pbark.parser.keywords.keyword_registry import KeywordRegistry


class PrintKeywordGroups:
    def __init__(self, registry: KeywordRegistry) -> None:
        self.print_words = registry.heard_words(
            "say", "says", "shout", "shouts", "mention", "mentions", "command", "commands",
            "announce", "announces", "declare", "declares", "tell", "tells", "whisper", "whispers",
            "mutter", "mutters", "squeak", "squeaks", "pant", "pants", "bark", "barks", "yap", "yaps",
            "growl", "growls", "whimper", "whimpers", "howl", "howls", "woof", "woofs", "whine",
            "whines", "whining", "yapping", "yapps",
        )
        self.exit_words = registry.heard_words("escape", "leave", "depart", "scram", "wander")
        self.stdin_words = registry.heard_words(
            "listen", "listens", "listening", "sniff", "sniffs", "perk", "perks", "hear", "hears",
            "eavesdrop", "eavesdrops", "ask", "asks", "prompt", "prompts", "await", "awaits",
        )
        self.wait_words = registry.heard_words(
            "wait", "waits", "waiting", "sleep", "sleeps", "sleeping", "nap", "naps", "napping",
            "doze", "dozes", "dozing", "snooze", "snoozes", "snoozing", "settle", "settles",
            "settling", "rest", "rests", "resting", "lounge", "lounges", "lounging", "pause",
            "pauses", "pausing", "linger", "lingers", "lingering", "loiter", "loiters", "loitering",
            "dream", "dreams", "dreaming",
        )

File: src/main/java/dev/klomptech/jbark/parser/keywords/PrintKeywordGroups.java
package dev.klomptech.jbark.parser.keywords;

import java.util.List;

public final class PrintKeywordGroups {

  public final List<String> printWords;
  public final List<String> exitWords;
  public final List<String> stdinWords;
  public final List<String> waitWords;

  public PrintKeywordGroups(final KeywordRegistry registry) {
    printWords =
        registry.heardWords(
            "say",
            "says",
            "shout",
            "shouts",
            "mention",
            "mentions",
            "command",
            "commands",
            "announce",
            "announces",
            "declare",
            "declares",
            "tell",
            "tells",
            "whisper",
            "whispers",
            "mutter",
            "mutters",
            "squeak",
            "squeaks",
            "pant",
            "pants",
            "bark",
            "barks",
            "yap",
            "yaps",
            "growl",
            "growls",
            "whimper",
            "whimpers",
            "howl",
            "howls",
            "woof",
            "woofs",
            "whine",
            "whines",
            "whining",
            "yapping",
            "yapps");
    exitWords = registry.heardWords("escape", "leave", "depart", "scram", "wander");
    stdinWords =
        registry.heardWords(
            "listen",
            "listens",
            "listening",
            "sniff",
            "sniffs",
            "perk",
            "perks",
            "hear",
            "hears",
            "eavesdrop",
            "eavesdrops",
            "ask",
            "asks",
            "prompt",
            "prompts",
            "await",
            "awaits");
    waitWords =
        registry.heardWords(
            "wait",
            "waits",
            "waiting",
            "sleep",
            "sleeps",
            "sleeping",
            "nap",
            "naps",
            "napping",
            "doze",
            "dozes",
            "dozing",
            "snooze",
            "snoozes",
            "snoozing",
            "settle",
            "settles",
            "settling",
            "rest",
            "rests",
            "resting",
            "lounge",
            "lounges",
            "lounging",
            "pause",
            "pauses",
            "pausing",
            "linger",
            "lingers",
            "lingering",
            "loiter",
            "loiters",
            "loitering",
            "dream",
            "dreams",
            "dreaming");
  }
}
